### DISCUSSION TITLE
state, county tag not available for all documents.

### DISCUSSION THREAD

**Discussion by mb12** [komoot/photon#803] [Q&A]
Can someone please suggest a workaround or a solution for the following?

For a lot of well known places, the state and county tags are not dumped in the photon database. I have verified it manually by writing an ES query for that osm_id in Kibana.  I have also verified that there isn't any state, county information for that osm_id when I query it using command line in nominatim.
Here is one specific example. Its for osm_id=4039486. Its for "Lake Superior"  in US. Is it possible to dump the county and state tags for the corresponding document in the photon database? 

I understand that this is a multipolygon geometry spanning multiple counties, states.  For those cases, it is perfectly fine to add any county, state. Preferably, the same logic that  is used for roads/streets spanning multiple administrative regions. For the specific example, I have included nominatim dumps it as "addresstype": "water",.

From a user's point of view, it very easy to  disambiguate between names if county, state information  can be appended for displaying. E.g. Bass Lake, Wisconsin vs Bass Lake, Michigan vs Bass Lake,  California.

I have also included the specific result from nominatim query search below that I have referenced above.
<pre>
nominatim search --query 'Lake Superior' | less
{
        "place_id": 46758114,
        "licence": "Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright",
        "osm_type": "relation",
        "osm_id": 4039486,
        "lat": "47.714434600000004",
        "lon": "-88.21163556659235",
        "category": "natural",
        "type": "water",
        "place_rank": 22,
        "importance": 0.5320626804991205,
        "addresstype": "water",
        "name": "Lake Superior",
        "display_name": "Lake Superior, United States",
        "boundingbox": [
            "46.4102412",
            "49.0186830",
            "-92.2727700",
            "-84.3543804"
        ]
    }
</pre>

**Comment by lonvia** [comment]
This is not really a Photon issue. The address information is already not added to these places by Nominatim. This is [by design](https://github.com/osm-search/Nominatim/pull/2038) and depends on the size of the object. Smaller lakes like the various Bass Lakes do get the county and state information, medium-sized ones will still have a state and the really large ones like the Great Lakes just get the country.

If you don't like that you need to change the source code of Nominatim and essentially revert the PR linked above for your own instance.

**Comment by mb12** [comment]
You are right about lake Superior.
A more appropriate example for the issue I am describing would be Lake Shasta (osm_id=1328693) 

1.) The nominatim search results  include "county" and "state" address tags. But they are not propagated to photon. 

https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=1328693&class=natural

2.) For this the photon document does not have "county" or "state" tags.
They can be computed by joining the placex and  place_addressline tables.
(getAddresses method in NominatimConnector).
That code never kicks in because rankAddress for this is 0. 

(To verify that this is indeed the case, I made the code change in NominatimConnecctor::getAddresses for specifically this osm_id. With that change the photon document got the address tags. I am not sure what is the right code change to make address tags consistent between photon and nominatim).

**Comment by lonvia** [comment]
I see what you mean. Sorry for the confusion. "Lake Superiour" was the one example where the bahviour was right. ;)

This is indeed a bug. And I'm really surprised nobody noticed for so long. The address parts will also be missing for the search (i.e. "Lake Shasta, California" won't come up with the lake at all). Fix pending in #808.
